### PR TITLE
Complete insert statement

### PR DESCRIPTION
**Pull request by soulasuna**
Complete insert statement

For #11661

Changes proposed in this pull request:
-  Add shadow statement validator.
-  Optimize the xml configuration tag name.
-  Add future shadow example.

### CODE REVIEW

File: examples/shardingsphere-jdbc-example/other-feature-example/future-shadow-example/future-shadow-spring-namespace-mybatis-example/src/main/java/org/apache/shardingsphere/example/shadow/spring/namespace/mybatis/ShadowSpringNamespaceMybatisExample.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+import org.apache.shardingsphere.example.shadow.spring.namespace.mybatis.service.ShadowService;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.context.support.ClassPathXmlApplicationContext;
+
+import java.sql.SQLException;
+import java.util.Collection;
+import java.util.LinkedList;
+
+public class ShadowSpringNamespaceMybatisExample {
```
Comment (tristaZero) [inline examples/shardingsphere-jdbc-example/other-feature-example/future-shadow-example/future-shadow-spring-namespace-mybatis-example/src/main/java/org/apache/shardingsphere/example/shadow/spring/namespace/mybatis/ShadowSpringNamespaceMybatisExample.java:28]: final one?

File: examples/shardingsphere-jdbc-example/other-feature-example/future-shadow-example/pom.xml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        <version>5.0.0-RC1-SNAPSHOT</version>
+    </parent>
+    <artifactId>future-shadow-example</artifactId>
+    <packaging>pom</packaging>
+    <name>${project.artifactId}</name>
+
+    <modules>
+        <module>future-shadow-spring-namespace-mybatis-example</module>
+    </modules>
+</project>
```
Comment (tristaZero) [inline examples/shardingsphere-jdbc-example/other-feature-example/future-shadow-example/pom.xml:35]: Here a new line needs.

File: shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/main/java/org/apache/shardingsphere/shadow/route/future/engine/determiner/algorithm/ColumnShadowAlgorithmDeterminer.java [hunk 3]
```diff
@@ -41,22 +42,22 @@
...
         Collection<String> columnNames = insertStatementContext.getInsertColumnNames();
-        List<List<Object>> groupedParameters = insertStatementContext.getGroupedParameters();
         Iterator<String> columnNamesIt = columnNames.iterator();
-        Iterator<List<Object>> groupedParametersIt = groupedParameters.iterator();
-        int count = Math.min(columnNames.size(), groupedParameters.size());
-        for (int i = 0; i < count; i++) {
-            Optional<Collection<PreciseColumnShadowValue<Comparable<?>>>> preciseColumnShadowValues = getPreciseColumnShadowValues(columnNamesIt.next(), groupedParametersIt.next(), tableName);
-            if (preciseColumnShadowValues.isPresent()) {
-                if (isPassColumn(preciseColumnShadowValues.get(), columnShadowAlgorithm, relatedShadowTables)) {
+        final List<InsertValueContext> insertValueContexts = insertStatementContext.getInsertValueContexts();
```
Comment (tristaZero) [inline shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/main/java/org/apache/shardingsphere/shadow/route/future/engine/determiner/algorithm/ColumnShadowAlgorithmDeterminer.java:46]: Why is a final definition needed?

File: examples/shardingsphere-jdbc-example/other-feature-example/future-shadow-example/future-shadow-spring-namespace-mybatis-example/pom.xml [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+            <groupId>org.apache.shardingsphere.example</groupId>
+            <artifactId>example-spring-mybatis</artifactId>
+            <version>${project.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.shardingsphere</groupId>
+            <artifactId>shardingsphere-jdbc-core-spring-namespace</artifactId>
+        </dependency>
+    </dependencies>
+</project>
```
Comment (tristaZero) [inline examples/shardingsphere-jdbc-example/other-feature-example/future-shadow-example/future-shadow-spring-namespace-mybatis-example/pom.xml:42]: Here a new line needs

**Review by tristaZero — CHANGES_REQUESTED** [review]

File: examples/shardingsphere-jdbc-example/other-feature-example/future-shadow-example/future-shadow-spring-namespace-mybatis-example/src/main/java/org/apache/shardingsphere/example/shadow/spring/namespace/mybatis/ShadowSpringNamespaceMybatisExample.java (hunk 1 above)
Comment (soulasuna) [reply examples/shardingsphere-jdbc-example/other-feature-example/future-shadow-example/future-shadow-spring-namespace-mybatis-example/src/main/java/org/apache/shardingsphere/example/shadow/spring/namespace/mybatis/ShadowSpringNamespaceMybatisExample.java:28]: I will fix it.

File: examples/shardingsphere-jdbc-example/other-feature-example/future-shadow-example/pom.xml (hunk 2 above)
Comment (soulasuna) [reply examples/shardingsphere-jdbc-example/other-feature-example/future-shadow-example/pom.xml:35]: I will fix it.

File: shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/main/java/org/apache/shardingsphere/shadow/route/future/engine/determiner/algorithm/ColumnShadowAlgorithmDeterminer.java (hunk 3 above)
Comment (soulasuna) [reply shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/main/java/org/apache/shardingsphere/shadow/route/future/engine/determiner/algorithm/ColumnShadowAlgorithmDeterminer.java:46]: I will fix it.

File: examples/shardingsphere-jdbc-example/other-feature-example/future-shadow-example/future-shadow-spring-namespace-mybatis-example/pom.xml (hunk 4 above)
Comment (soulasuna) [reply examples/shardingsphere-jdbc-example/other-feature-example/future-shadow-example/future-shadow-spring-namespace-mybatis-example/pom.xml:42]: I will fix it.

File: examples/shardingsphere-jdbc-example/other-feature-example/future-shadow-example/future-shadow-spring-namespace-mybatis-example/src/main/java/org/apache/shardingsphere/example/shadow/spring/namespace/mybatis/domain/OrderInfo.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+package org.apache.shardingsphere.example.shadow.spring.namespace.mybatis.domain;
+
+import lombok.Data;
+
+import java.io.Serializable;
+
+@Data
+public final class OrderInfo implements Serializable {
+
```
Comment (taojintianxia) [inline examples/shardingsphere-jdbc-example/other-feature-example/future-shadow-example/future-shadow-spring-namespace-mybatis-example/src/main/java/org/apache/shardingsphere/example/shadow/spring/namespace/mybatis/domain/OrderInfo.java:26]: why not add a serialVersionUID ?

Comment (soulasuna) [reply examples/shardingsphere-jdbc-example/other-feature-example/future-shadow-example/future-shadow-spring-namespace-mybatis-example/src/main/java/org/apache/shardingsphere/example/shadow/spring/namespace/mybatis/domain/OrderInfo.java:26]: I will add serialVersionUID.

File: examples/shardingsphere-jdbc-example/other-feature-example/future-shadow-example/future-shadow-spring-namespace-mybatis-example/src/main/resources/META-INF/mappers/ShadowMapper.xml [hunk 6]
```diff
@@ -0,0 +1,30 @@
+<?xml version="1.0" encoding="UTF-8" ?>
```
Comment (taojintianxia) [inline examples/shardingsphere-jdbc-example/other-feature-example/future-shadow-example/future-shadow-spring-namespace-mybatis-example/src/main/resources/META-INF/mappers/ShadowMapper.xml:1]: why there is a whitespace  "UTF-8" ?

Comment (soulasuna) [reply examples/shardingsphere-jdbc-example/other-feature-example/future-shadow-example/future-shadow-spring-namespace-mybatis-example/src/main/resources/META-INF/mappers/ShadowMapper.xml:1]: This format is consistent with the style of the entire example.

**Review by tristaZero — APPROVED** [review]

**Pull request merged**